app/models/customer.rb

```diff
@@ -11,4 +11,13 @@ def self.top_5
   def count_success_transactions
     transactions.where("transactions.result = 0").count
   end
+
+  def self.top_customers(merchant_id)
+    joins(invoices: { invoice_items: { item: :merchant }, transactions: :invoice })
+    .where('merchants.id = ?', merchant_id)
+    .where("transactions.result = 0")
```

Comment:
This query looks like a beast and is awesome, thank you for getting this one complete!